apps/backend/src/utils/ssh_command_manager.py

```
@@ -0,0 +1,509 @@
+            )
+        
+        # Format command with parameters
+        parameters = parameters or {}
+        try:
+            formatted_command = command_def.command_template.format(**parameters)
+        except KeyError as e:
+            raise SSHCommandError(
+                f"Missing parameter for command {command_name}: {e}",
+                command=command_def.command_template,
+                hostname=connection_info.host
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)